adapter/statsd/statsd.go

```diff
@@ -154,31 +154,19 @@ func (a *aspect) record(value adapter.Value) error {
 		buf.Reset()
 		bufferPool.Put(buf)
 	}
-	switch value.Kind {
-	case adapter.Gauge:
+	if value.Definition.Kind == adapter.Gauge {
 		v, err := value.Int64()
 		if err != nil {
 			return fmt.Errorf("could not record gauge '%s' with err: %s", mname, err)
 		}
 		return a.client.Gauge(mname, v, a.rate)
-	case adapter.Counter:
```

Comment:
I don't think we want to remove this switch. While it's simpler today since we only support two kinds of metrics, in the (near?) future we'll be adding more kinds of metrics - whoever adds support for those is going to have to come in and reimplement the switch.